src/git.ts

```diff
@@ -154,18 +154,29 @@ export function pull(withRebase = true, cwd?: string) {
   return gitWithNetwork(args, cwd);
 }
 
-async function git(args: string[], cwd: string | undefined) {
-  const result = await spawn('git', args, { stdio: 'pipe', cwd: cwd ?? projectPath });
-  return result.stdout.trim();
-}
+// git on some Docker images such as alpine node behaves very odd when you run multiple git command
+// in parallel. This ensures that we're only running a single git command at a time.
+let currentCommand: Promise<unknown> | undefined;
 
-function gitSync(args: string[], cwd: string | undefined) {
-  const result = spawnSync('git', args, { stdio: 'pipe', cwd: cwd ?? projectPath });
-  return result.stdout.trim();
+async function git(args: string[], cwd: string | undefined) {
+  if (currentCommand) {
+    await currentCommand;
```

Comment:
Wouldn't this just allow the first git command to be run separately, and then the rest all at once? :thinking: Or at least all the ones that were called before the first `currentCommand` has resolved.

Reply:
That's very true! Nice catch, brain fart from me there. 😄 Fixed now!